Backpack/CardButton/Classes/SaveCardButton.swift

```diff
@@ -0,0 +1,273 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+public class BPKSaveCardButton: UIButton {
+
+    public var checked: Bool {
+        didSet {
+            updateLookAndFeel()
+            onCheckedChange?(checked)
+        }
+    }
+
+    public var style: BPKCardButtonStyle {
+        didSet { updateLookAndFeel() }
+    }
+
+    public var size: BPKCardButtonSize {
+        didSet { updateLookAndFeel() }
+    }
+
+    public var onCheckedChange: ((Bool) -> Void)?
+
+    private let viewConfigurator = CardButtonViewConfigurator()
+    private let containedBackgroundCircle: UIView
+
+    public init(
+        checked: Bool = false,
+        accessibilityLabel: String,
+        style: BPKCardButtonStyle = .default,
+        size: BPKCardButtonSize = .default,
+        onCheckedChange: ((Bool) -> Void)? = nil
```

Comment:
We could have a default tap gesture like we did on the BPKChip and then communicate valueChanged whenever the button is tapped? You can see how the chip handles this.

Reply:
@frugoman  suggested me to remove the onClick callback because customer already has access to iOS's `addTarget` api. [(link)](https://github.com/Skyscanner/backpack-ios/pull/1564#discussion_r1098936550)

I prefer to skip the extra tapGuesture